x/nft/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,583 @@
+		},
+		{
+			"Success",
+			func(require *require.Assertions) {
+				suite.TestMint()
+				req = &nft.QueryBalanceRequest{
+					ClassId: testClassID,
+					Owner:   suite.addrs[0].String(),
+				}
+				balance = 1
+			},
+			"",
+			func(require *require.Assertions, res *nft.QueryBalanceResponse) {
+				require.Equal(res.Amount, balance)
+			},
+		},
+	}
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			require := suite.Require()
+			tc.malleate(require)
+			result, err := suite.queryClient.Balance(gocontext.Background(), req)
+			if tc.expError == "" {
+				require.NoError(err)
+			} else {
+				require.Error(err)
+				require.Contains(err.Error(), tc.expError)
+			}
+			tc.postTest(require, result)
+		})
+	}
+}
+
+func (suite *TestSuite) TestGRPCQueryOwner() {
+	var (
+		req   *nft.QueryOwnerRequest
+		owner string
+	)
+	testCases := []struct {
+		msg      string
+		malleate func(require *require.Assertions)
+		expError string
+		postTest func(require *require.Assertions, res *nft.QueryOwnerResponse)
+	}{
+		{
+			"fail empty ClassId",
+			func(require *require.Assertions) {
+				req = &nft.QueryOwnerRequest{
+					Id: testID,
+				}
+			},
+			"class id can not be empty",
+			func(require *require.Assertions, res *nft.QueryOwnerResponse) {},
+		},
+		{
+			"fail empty nft id",
+			func(require *require.Assertions) {
+				req = &nft.QueryOwnerRequest{
+					ClassId: testClassID,
+				}
+			},
+			"nft id can not be empty",
+			func(require *require.Assertions, res *nft.QueryOwnerResponse) {},
+		},
+		{
+			"success but nft id not exist",
+			func(require *require.Assertions) {
+				req = &nft.QueryOwnerRequest{
+					ClassId: testClassID,
+					Id:      "kitty2",
+				}
+			},
+			"",
+			func(require *require.Assertions, res *nft.QueryOwnerResponse) {
+				require.Equal(res.Owner, owner)
+			},
+		},
+		{
+			"success but class id not exist",
+			func(require *require.Assertions) {
+				req = &nft.QueryOwnerRequest{
+					ClassId: "kitty1",
+					Id:      testID,
+				}
+			},
+			"",
+			func(require *require.Assertions, res *nft.QueryOwnerResponse) {
+				require.Equal(res.Owner, owner)
+			},
+		},
+		{
+			"Success",
+			func(require *require.Assertions) {
+				suite.TestMint()
+				req = &nft.QueryOwnerRequest{
+					ClassId: testClassID,
+					Id:      testID,
+				}
+				owner = suite.addrs[0].String()
+			},
+			"",
+			func(require *require.Assertions, res *nft.QueryOwnerResponse) {
+				require.Equal(res.Owner, owner)
+			},
+		},
+	}
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			require := suite.Require()
+			tc.malleate(require)
+			result, err := suite.queryClient.Owner(gocontext.Background(), req)
+			if tc.expError == "" {
+				require.NoError(err)
+			} else {
+				require.Error(err)
+				require.Contains(err.Error(), tc.expError)
+			}
+			tc.postTest(require, result)
```

Comment:
we don't know which test will fail if we pass `require` this way. Which is harder to debug if the tests will start failing. Let's add an `index` to the `postTest` and `mallate` functions, and use it as a message argument when we call require.